src/processing/permits_clustering.py

```
@@ -0,0 +1,764 @@
+                ))
+                update_count += 1
+                
+                # Commit in batches for performance
+                if update_count % 100 == 0:
+                    conn.commit()
+            
+            # Final commit
+            conn.commit()
+            logger.info(f"Successfully updated {update_count} permits with cluster assignments")
+            
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)